stan/math/prim/fun/rep_array.hpp

```diff
@@ -8,28 +8,32 @@
 namespace stan {
 namespace math {
 
-template <typename T>
-inline std::vector<T> rep_array(const T& x, int n) {
+template <typename In>
+inline std::vector<plain_type_t<In>> rep_array(const In& x, int n) {
+  using T = plain_type_t<In>;
```

Comment:
```suggestion
template <typename In, typename T = plain_type_t<In>>
inline std::vector<T> rep_array(const In& x, int n) {
```

Reply:
May also be more readable to use a typename like ```PlainT``` or ```T_plain```

Reply:
I personally findcode more readable if we don't cram to many things into the line with templates. In this function it would not matter, but we have some functions with a million requires, which would get messy. I'd like to keep all functions consistent.

If you disagree I can still change this.

Reply:
No that's fine, I'm not particularly married to it